### PR TITLE
Refactor multiples logcollector test

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
|      [3218](https://github.com/wazuh/wazuh-qa/issues/3218)   |

## Description

This PR aims to fix the following tests reported as unstable:
 - `test_logcollector/test_configuration/test_basic_configuration_age.py`
 - `test_logcollector/test_configuration/test_basic_configuration_query.py`
 - `test_logcollector/test_command_monitoring/test_command_execution_freq.py `
 - `test_logcollector/test_configuration/test_basic_configuration_only_future_events.py`
 - `test_logcollector/test_command_monitoring/test_command_execution_dbg.py`
 
A full refactor of the test with the new QA structure has been carried on.

### Added

For `test_command_execution_freq` it has added a fixture that modifies the date format of the Windows OS to use the `TimeMachine.travel_to_future` function. 

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @fedepacher  (Developer)  |           |  :green_circle::green_circle::green_circle:  | :green_circle::green_circle::green_circle: |   Windows/Linux      |         | Nothing to highlight |
| @Rebits  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

Jenkins Launches:
- [Launch 1](https://devel.ci.wazuh.info/job/Test_integration/1704/)
- [Launch 2](https://devel.ci.wazuh.info/job/Test_integration/1708/)
- [Launch 3](https://devel.ci.wazuh.info/job/Test_integration/1709/)

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from wazuh_testing import T_30, T_60
+from wazuh_testing.modules.logcollector import LOG_COLLECTOR_PREFIX, MACOS_LOG_COMMAND_PATH
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.tools import LOG_FILE_PATH
+
+
+def make_logcollector_callback(pattern, prefix=LOG_COLLECTOR_PREFIX, escape=False):
+    """Create a callback function from a text pattern.
+
+    It already contains the analsisd prefix.
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py:12]: Change it to logcollector

File: deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    Args:
+        pattern (str): String to match on the log.
+        prefix (str): regular expression used as a prefix before the pattern.
+        escape (bool): Flag to escape special characters in the pattern
+
+    Returns:
+        lambda: function that returns if there's a match in the file
+
+    Examples:
+        >>> callback_bionic_update_started = make_vuln_callback("Starting Ubuntu Bionic database update")
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py:23]: Update example to use logcollector

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py [hunk 3]
```diff
@@ -59,99 +59,61 @@
...
+
+if sys.platform != 'linux':
+    index = 0
+    for _ in range(len(t1_configuration_metadata)):
+        if t1_configuration_metadata[index]['os'] == 'linux' and index > 0:
+            index = remove_item_from_list(index)
+        index += 1
+
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
```
Comment (Rebits) [inline tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py:116]: According to this logic block:
- Do the non-Linux OS launch all cases?
- Maybe is better to use a test case file for each OS (according to the test differences)

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py [hunk 4]
```diff
@@ -56,60 +56,70 @@
...
 
 local_internal_options = {'logcollector.remote_commands': '1', 'logcollector.debug': '2', 'monitord.rotate_log': '0',
                           'windows.debug': '2'}
 
+if sys.platform == 'win32':
+    location = r'C:\testing\file.txt'
+    prefix = WINDOWS_AGENT_PREFIX
+
+
+@pytest.fixture(scope='module')
```
Comment (Rebits) [inline tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py:100]: Move to logcollector module conftest

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py [hunk 5]
```diff
@@ -56,194 +56,144 @@
...
+    prefix = WINDOWS_AGENT_PREFIX
+    daemon = 'wazuh-agent.exe'
 
 
-if sys.platform == 'win32':
-    prefix = WINDOWS_AGENT_DETECTOR_PREFIX
-    log_format_list += ['eventchannel']
-elif sys.platform == 'darwin':
-    log_format_list += ['macos']
+def remove_item_from_list(item):
```
Comment (Rebits) [inline tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py:100]: Same as before. If each os has its own cases mybe is better to create a test_case file for each one.

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_age.py [hunk 6]
```diff
@@ -221,25 +178,24 @@ def test_configuration_age(get_configuration, configure_environment):
...
                 pytest.xfail("Windows agent allows invalid localfile configuration:\
                               https://github.com/wazuh/wazuh/issues/10890")
-                expected_exception = ValueError
             else:
-                expected_exception = sb.CalledProcessError
-
-            with pytest.raises(expected_exception):
-                control_service('start', daemon=LOGCOLLECTOR_DAEMON)
-                check_configuration_age_invalid(cfg)
+                try:
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_age.py:195]: Change to with pytest.raises logic

**Review by Rebits — CHANGES_REQUESTED** [review]
Good job, but some changes are requied

File: deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py (hunk 1 above)
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py:12]: Done in [here]([40e7276](https://github.com/wazuh/wazuh-qa/pull/3486/commits/40e727636c5f3938a6430afe75851cecd436d81c))

File: deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py (hunk 2 above)
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py:23]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/40e727636c5f3938a6430afe75851cecd436d81c)

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py (hunk 3 above)
Comment (fedepacher) [reply tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py:116]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/fa745773ca16d8f5751f9a9f5d0d3c4c3ccc40a1)

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py (hunk 4 above)
Comment (fedepacher) [reply tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py:100]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/21f9ee85c4d9203435d6adca8336956810f20708)

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_age.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (fedepacher) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_age.py:195]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/78c9c6c5df5cbad28da01a70cde48a1b9907a2a8)

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py (hunk 5 above)
Comment (fedepacher) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py:100]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/5ad95d105667db211f44fcdd35ffcb640efe9d08)

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py [hunk 7]
```diff
@@ -56,194 +56,144 @@
...
+        # remove windows and macos cases
+        index = remove_item_from_list(index)
+    index += 1
+
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+
+
+@pytest.fixture(scope="function")
+def generate_macos_logs(metadata):
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py:145]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/5ad95d105667db211f44fcdd35ffcb640efe9d08)

**Pull request closed without merging**